Tools/CMake/AMReXSetDefines.cmake

```diff
@@ -186,3 +186,11 @@ if(WIN32 AND BUILD_SHARED_LIBS)
   add_amrex_define(AMREX_IS_DLL NO_LEGACY)
   target_compile_definitions( amrex PRIVATE AMREX_IS_DLL_BUILDING)
 endif()
+
+#
+# Windows math symbols <cmath>: M_PI
+# https://docs.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170
+#
+if(WIN32)
+  target_compile_definitions(amrex PUBLIC _USE_MATH_DEFINES)
```

Comment:
Would it be better to do this in `AMReX_Geometry.H`?

Reply:
I think not, I want to avoid that some TUs are translated with the flag and some without it. That usually means trouble, since I cannot guarantee that they do not do other things in that header that would violate the ODR:
- https://en.cppreference.com/w/cpp/language/definition

Also, setting this generally reduced a lot of Windows-specific surprises and feels more like Linux in general (similar to what we set for the extra C++ keywords `and`, `or`, etc.).

Reply:
I don't see why this has anything to do with ODR.  I think you misunderstood me.  I am suggesting add
```
#ifdef _WIN32
#define _USE_MATH_DEFINES
#endif
```
to `AMReX_Geometry.H` before `#include <cmath>` instead of trusting cmake.

Reply:
Maybe we should avoid all the trouble with `M_PI` and define our own `constexpr Real pi = 3.141592653589793238462643383279502884197` in that one function.

Reply:
I guess my suggestion has a flaw.
```
#include "foo.H"  // which include cmath
#define _USE_MATH_DEFINES  /* won't work because of include guard in cmath */
#include <cmath>
```

I think we should define our own pi in that function.